wechaty-puppet/helper/async.go

```diff
@@ -38,10 +39,14 @@ func NewAsync(maxWorkerNum int) IAsync {
 	}
 	return &async{
 		maxWorkerNum: maxWorkerNum,
+		wg:           sync.WaitGroup{},
 	}
 }
 
 func (a *async) AddTask(task Task) {
+	if a.tasks == nil {
+		a.tasks = make([]Task, 0)
+	}
```

Comment:
There is no need to judge whether `slice` is `nil`, because golang is very smart, see: https://play.golang.org/p/azt9SnSknTQ

Reply:
Cool, I didn't know go is so smart.